common/metrics/metricstest/metricstest_test.go

```diff
@@ -42,23 +42,29 @@ func TestBasic(t *testing.T) {
 		metrics.StringTag("l2", "v2"),
 		metrics.StringTag("l1", "v1"),
 	}
+	expectedSystemTags := []metrics.Tag{
+		metrics.StringTag("otel_scope_name", "temporal"),
+		metrics.StringTag("otel_scope_version", ""),
+	}
+	expectedCounterTags := append(expectedSystemTags, counterTags...)
 	counter := handler.WithTags(counterTags...).Counter(counterName)
 	counter.Record(1)
 	counter.Record(1)
 
 	s1 := handler.MustSnapshot()
-	require.Equal(t, float64(2), s1.MustCounter(counterName, counterTags...))
+	require.Equal(t, float64(2), s1.MustCounter(counterName+"_total", expectedCounterTags...))
```

Comment:
Is this new `_total` suffix always added by otel, or is it configurable?

I wondering if this implies that enabling otel means that all counter based metrics, say in alerts or dashboard definitions, will break.

Reply:
I think this is a always thing. https://github.com/open-telemetry/opentelemetry-go/blob/22434311ae4783010a04081634ec503d9cbb5e33/exporters/prometheus/exporter.go#L72

Reply:
For users use otel, I think they will aware of this. For users switch from tally to otel, unfortunately, queries have to be updated.

Reply:
Wouldn't hurt to double check that our metrics are currently being exported that way via otel and we're not introducing a breaking change, just in case.

Reply:
This is a breaking change in otel.

Reply:
Does this change metric names for only the otel plugin, or all metrics plugins?

Reply:
Only item. No breaking change in Tally.

Reply:
That's good, is there anything we need to do in this PR to note the breaking change so that users will be aware of it? It'll be a very important aspect of users upgrade path to ensure their dashboards and alerts continue working.

Reply:
I will update our 1.20 release note.